test/jdk/java/nio/channels/Channels/TransferTo.java

```diff
@@ -59,59 +66,90 @@
 
     private static final Random RND = RandomFactory.getRandom();
 
-    public static void main(String[] args) throws Exception {
-        test(fileChannelInput(), fileChannelOutput());
-        test(readableByteChannelInput(), defaultOutput());
-    }
+    @DataProvider
+    public static Object[][] streamCombinations() throws Exception {
+        return new Object[][] {
+            // tests FileChannel.transferTo(FileChannel) optimized case
+            { fileChannelInput(), fileChannelOutput() },
 
-    private static void test(InputStreamProvider inputStreamProvider, OutputStreamProvider outputStreamProvider)
-            throws Exception {
-        testNullPointerException(inputStreamProvider);
-        testStreamContents(inputStreamProvider, outputStreamProvider);
+            // tests InputStream.transferTo(OutputStream) default case
+            { readableByteChannelInput(), defaultOutput() }
+        };
     }
 
-    private static void testNullPointerException(InputStreamProvider inputStreamProvider) throws Exception {
-        try (InputStream in = inputStreamProvider.input()) {
-            assertThrowsNPE(() -> in.transferTo(null), "out");
-        }
+    /**
+     * Testing API compliance: Input stream must throw NullPointerException when parameter "out" is null.
+     */
+    @Test(dataProvider = "streamCombinations")
+    public void testNullPointerException(InputStreamProvider inputStreamProvider,
+            OutputStreamProvider outputStreamProvider) throws Exception {
+        // tests empty input stream
+        assertThrows(NullPointerException.class, () -> inputStreamProvider.input().transferTo(null));
 
-        try (InputStream in = inputStreamProvider.input((byte) 1)) {
-            assertThrowsNPE(() -> in.transferTo(null), "out");
-        }
+        // tests single-byte input stream
+        assertThrows(NullPointerException.class, () -> inputStreamProvider.input((byte) 1).transferTo(null));
 
-        try (InputStream in = inputStreamProvider.input((byte) 1, (byte) 2)) {
-            assertThrowsNPE(() -> in.transferTo(null), "out");
-        }
+        // tests dual-byte input stream
+        assertThrows(NullPointerException.class, () -> inputStreamProvider.input((byte) 1, (byte) 2).transferTo(null));
     }
 
-    private static void testStreamContents(InputStreamProvider inputStreamProvider,
+    /**
+     * Testing API compliance: Complete content of input stream must be transferred to output stream.
+     */
+    @Test(dataProvider = "streamCombinations")
+    public void testStreamContents(InputStreamProvider inputStreamProvider,
             OutputStreamProvider outputStreamProvider) throws Exception {
+        // tests empty input stream
         checkTransferredContents(inputStreamProvider, outputStreamProvider, new byte[0]);
+
+        // tests input stream with a lenght between 1k and 4k
```

Comment:
/lenght/length/

Reply:
Good catch! I will fix that.

Reply:
Done by https://github.com/openjdk/jdk/pull/5421/commits/efa8ace1db80ed51006948f201675d224fd14dea